e2e-tests/batch-mode-validation.sh

```diff
@@ -0,0 +1,175 @@
+  print_message "Total patients synced to parquet ---> ${total_patients_streamed}"
+  total_encounters_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ${HOME_PATH}/${TEST_TYPE}/Encounter/ | awk '{print $3}')
+  print_message "Total encounters synced to parquet ---> ${total_encounters_streamed}"
+  total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ${HOME_PATH}/${TEST_TYPE}/Observation/ | awk '{print $3}')
+  print_message "Total obs synced to parquet ---> ${total_obs_streamed}"
+
+  if [[ ${total_patients_streamed} == ${TOTAL_TEST_PATIENTS} && ${total_encounters_streamed} \
+        == ${TOTAL_TEST_ENCOUNTERS} && ${total_obs_streamed} == ${TOTAL_TEST_OBS} ]] \
+    ; then
+    print_message "BATCH MODE WITH PARQUET SINK EXECUTED SUCCESSFULLY USING ${TEST_TYPE} MODE"
+  else
+    print_message "BATCH MODE WITH PARQUET SINK TEST FAILED USING ${TEST_TYPE} MODE"
+    exit 1
+  fi
+}
+
+#######################################
+# Function that counts resources in  FHIR server
+# and compares output to what is in openmrs server
+# Globals:
+#   HOME_PATH
+#   TEST_TYPE
+#   SINK_SERVER
+#   TOTAL_TEST_PATIENTS
+#   TOTAL_TEST_ENCOUNTERS
+#   TOTAL_TEST_OBS
+#######################################
+function test_fhir_sink() {
+
+  print_message "Finding number of patients, encounters and obs in FHIR server"
+
+  mkdir ${HOME_PATH}/fhir
+  curl -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 \
+    ${SINK_SERVER}/fhir/Patient/?_summary=count 2>/dev/null >>${HOME_PATH}/fhir/patients.json
+
+  curl -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 \
+    ${SINK_SERVER}/fhir/Encounter/?_summary=count 2>/dev/null >>${HOME_PATH}/fhir/encounters.json
+
+  curl -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 \
+    ${SINK_SERVER}/fhir/Observation/?_summary=count 2>/dev/null >>${HOME_PATH}/fhir/obs.json
+
+  print_message "Counting number of patients, encounters and obs sinked to fhir files"
+
+  total_patients_sinked_fhir=$(jq '.total' ${HOME_PATH}/fhir/patients.json)
+  print_message "Total patients sinked to fhir ---> ${total_patients_sinked_fhir}"
+
+  total_encounters_sinked_fhir=$(jq '.total' ${HOME_PATH}/fhir/encounters.json)
+  print_message "Total encounters sinked to fhir ---> ${total_encounters_sinked_fhir}"
+
+  total_obs_sinked_fhir=$(jq '.total' ${HOME_PATH}/fhir/obs.json)
+  print_message "Total observations sinked to fhir ---> ${total_obs_sinked_fhir}"
+
+  if [[ ${total_patients_sinked_fhir} == ${TOTAL_TEST_PATIENTS} && ${total_encounters_sinked_fhir} \
+        == ${TOTAL_TEST_ENCOUNTERS} && ${total_obs_sinked_fhir} == ${TOTAL_TEST_OBS} ]] \
+    ; then
+    print_message "BATCH MODE WITH FHIR SERVER SINK EXECUTED SUCCESSFULLY USING ${TEST_TYPE} MODE"
+  else
+    print_message "BATCH MODE WITH FHIR SERVER SINK TEST FAILED USING ${TEST_TYPE} MODE"
+    exit 1
+  fi
+}
+
+setup $1 $2 $3
```

Comment:
Since the intention is to be able to run this locally outside the Cloud Build flow, please add some parameter validation and also a usage message that describes what each parameter is for.

Reply:
added a usage and validation function